shell/platform/linux/fl_keyboard_pending_event.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_keyboard_pending_event.h"
+
+/**
+ * FlKeyboardPendingEvent:
+ * A record for events that have been received by the handler, but
+ * dispatched to other objects, whose results have yet to return.
+ *
+ * This object is used by both the "pending_responds" list and the
+ * "pending_redispatches" list.
+ */
+
+struct _FlKeyboardPendingEvent {
+  GObject parent_instance;
+
+  // The target event.
+  FlKeyEvent* event;
+
+  // Unique ID to identify pending responds.
+  uint64_t sequence_id;
+
+  // The number of responders that haven't replied.
+  size_t unreplied;
+
+  // Whether any replied responders reported true (handled).
+  bool any_handled;
+
+  // A value calculated out of critical event information that can be used
+  // to identify redispatched events.
+  uint64_t hash;
+};
+
+G_DEFINE_TYPE(FlKeyboardPendingEvent, fl_keyboard_pending_event, G_TYPE_OBJECT)
+
+static void fl_keyboard_pending_event_dispose(GObject* object) {
+  FlKeyboardPendingEvent* self = FL_KEYBOARD_PENDING_EVENT(object);
+
+  g_clear_object(&self->event);
+
+  G_OBJECT_CLASS(fl_keyboard_pending_event_parent_class)->dispose(object);
+}
+
+static void fl_keyboard_pending_event_class_init(
+    FlKeyboardPendingEventClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_keyboard_pending_event_dispose;
+}
+
+static void fl_keyboard_pending_event_init(FlKeyboardPendingEvent* self) {}
+
+// Create a new FlKeyboardPendingEvent by providing the target event,
+// the sequence ID, and the number of responders that will reply.
```

Comment:
super pedantic nit: Flutter style would be "Creates [...]", but happy to defer to standard GTK style if this is more idiomatic.

We don't call it out specifically in our style guide, but tend to fall back to Effective Dart, and because we're relatively lazy, tend to do so for C/Obj-C/C++ etc. too. 

https://dart.dev/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs

Reply:
In existing comments we've always met the Dart style, so I'll update this.